src/asDate.ts

```diff
@@ -1,5 +1,16 @@
 import type { DateField, TimestampField } from "@prismicio/types";
 
+/**
+ * The return type of `asDate()`.
+ */
+type AsDateReturnType<
+	Field extends DateField | TimestampField | null | undefined,
+> = Field extends DateField<"filled">
+	? Date
+	: Field extends TimestampField<"filled">
```

Comment:
Small thing, but I think we can combine this to `Field extends DateField<"filled"> | TimestampField<"filled"> ? Date : null`

Reply:
Done! Refactored for date & links